internal/server/httpapi/trigger.go

```diff
@@ -0,0 +1,398 @@
+			w.WriteHeader(http.StatusOK)
+			flusher.Flush()
+
+			enc := json.NewEncoder(w)
+
+			// NOTE(briancain): We skip every two jobs here because when we call RunTrigger
+			// via gRPC, it eventually queues the trigger jobs through on-demand runners, and
+			// queueJobMulti returns three jobs: StartTask, the job to be queued, and StopTask. People
+			// really only expect output from the job to be queued, so we only stream that back.
+
+			// For example, a trigger that queues 2 Waypoint operation jobs returns 6 total job ids:
+			// Job List: [ {0: StartTask, 1: WP Operation 1, 2: StopTask}, {3: StartTask, 4: WP Operation 2, 5: StopTask}, ... ]
+
+			var triggerJobs []string
+			for i := 1; i < len(jobIds); i += 3 {
+				triggerJobs = append(triggerJobs, jobIds[i])
+			}
+
+			var (
+				wg sync.WaitGroup
+				mu sync.Mutex
+			)
+
+			log.Trace("starting job stream for jobs", "total_jobs", len(triggerJobs))
+			wg.Add(len(triggerJobs))
+
+			// NOTE(briancain): This loop starts N goroutines concurrently for
+			// each trigger job to stream back to the requester.
+			// TODO(briancain): turn the http errors into Messages to send back
+			for _, jId := range triggerJobs {
+				go func(jId string) {
+					defer wg.Done()
+
+					stream, err := client.GetJobStream(ctx, &pb.GetJobStreamRequest{
+						JobId: jId,
+					})
+					if err != nil {
+						log.Error("server failed to get job stream output for trigger", "job_id", jId, "err", err)
+						http.Error(w,
+							fmt.Sprintf("server failed to obtain job stream output: %s", err),
+							http.StatusInternalServerError)
+						return
+					}
+
+					// Wait for open confirmation
+					resp, err := stream.Recv()
+					if err != nil {
+						log.Error("server failed to stream job output", "err", err)
+						http.Error(w,
+							fmt.Sprintf("server failed to receive job stream output: %s", err),
+							http.StatusInternalServerError)
+						return
+					}
+					if _, ok := resp.Event.(*pb.GetJobStreamResponse_Open_); !ok {
+						log.Error("server failed to open job stream output, got unexpected message", "event", resp.Event)
+						http.Error(w,
+							fmt.Sprintf("job stream failed to open, got unexpected message: %T", resp.Event),
+							http.StatusInternalServerError)
+						return
+					}
+
+					var (
+						jobComplete bool
+						exitCode    string
+					)
+
+					log.Trace("reading job stream for job", "job_id", jId)
+
+					// read and send the stream
+					for {
+						resp, err := stream.Recv()
+						if err != nil {
+							http.Error(w,
+								fmt.Sprintf("server failed to receive job stream output: %s", err),
+								http.StatusInternalServerError)
+							return
+						}
+						if resp == nil {
+							// This shouldn't happen, but if it does, just ignore it.
+							log.Warn("nil response received, ignoring")
+							continue
+						}
+
+						select {
+						case <-cn.CloseNotify():
+							log.Trace("client closed connection to stream")
+							return
+						default:
```

Comment:
nit: since the bulk of the logic is in this `default:` block, we could move this `select` to the top like so:

```go
for {
  select {
    case <-cn.CloseNotify():
      log.Trace("client closed connection to stream")
      return
    default:
      // Get jobstream output and return string chunk message back
      time.Sleep(time.Second)
  }
  [ .. rest of the code .. ]
}
```

This format reduces the stack by 1 and I feel is just easier to reason about. It also moves the listening for the `CloseNotify()` to the first thing we do in the loop, instead of `Recv()`